build_test/lib/src/resolve_source.dart

```diff
@@ -192,15 +204,17 @@ typedef FutureOr<T> _ResolverAction<T>(Resolver resolver);
 class _ResolveSourceBuilder<T> implements Builder {
   final _ResolverAction<T> _action;
   final Future _tearDown;
+  final AssetId _resolverFor;
+
   final onDone = new Completer<T>();
 
-  _ResolveSourceBuilder(this._action, this._tearDown);
+  _ResolveSourceBuilder(this._action, this._resolverFor, this._tearDown);
 
   @override
   Future<Null> build(BuildStep buildStep) async {
     if (onDone.isCompleted) return;
     var result = await _action(buildStep.resolver);
-    if (!onDone.isCompleted) {
+    if (!onDone.isCompleted && _resolverFor == buildStep.inputId) {
       // With resolveSources (plural), this function is called multiple times
```

Comment:
Is this comment still true? I think we will only see a given `inputId` once so we can remove the other check and the comment.

Reply:
Good catch! Done.